HPCC-16858 Detect compressed files when reattaching XREF files.

jakesmith opened the pull request:
Check physical files, test to see if compressed so as to mark
the meta data correctly, also add size or compressedSize
dependent on whether compressed or not.

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

jakesmith commented:
@AttilaVamos - please review

HPCCSmoketest commented:
Automated Smoketest
Sha: 1ddf52b4e7b2def4fd702af2e79b79f84f88094b
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 90.878 seconds |

Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unittest result:

| total | passed | failed | timeout | 
|---|---|---|---|
|86 |86 |0 |0 |

HPCC Stop: OK
HPCC Uninstall: OK

AttilaVamos commented on system/jlib/jlzw.cpp:
```diff
@@ -2326,37 +2326,63 @@ class CCompressedFile : implements ICompressedFileIO, public CInterface
+        if (expander&&(trailer.recordSize!=0)) {
+            throw MakeStringException(-1, "Compressed file format error(%d), Encrypted?",trailer.recordSize);
+        }
+        unsigned compMethod1 = COMPRESS_METHOD_LZW;
```
Is there any meaning of numbering compMethod variable?

jakesmith replied:
No particular reason this temp variable is called 'compMethod*1*', no.
It was moved code, I don't know why it was called that.
I'll rename it.

AttilaVamos reviewed (approved):
Looks good. I have only one question.

AttilaVamos commented:
@jakesmith back to you.

jakesmith commented:
@AttilaVamos - renamed compMethod1 to compMethod.

Please check , then I'll squash.

HPCCSmoketest commented:
Automated Smoketest
Sha: 81aaac126f629550a5211583cde9f1cb1bea9e01
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.757 seconds |

Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unittest result:

| total | passed | failed | timeout | 
|---|---|---|---|
|86 |86 |0 |0 |

HPCC Stop: OK
HPCC Uninstall: OK

AttilaVamos commented:
@jakesmith it seems good to merge.

jakesmith commented:
Squashed.

@richardkchapman - ready to merge.

HPCCSmoketest commented:
Automated Smoketest
Sha: 70a360f93079661e645189f142407f904fdc0d49
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 90.425 seconds |

Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unittest result:

| total | passed | failed | timeout | 
|---|---|---|---|
|86 |86 |0 |0 |

HPCC Stop: OK
HPCC Uninstall: OK

richardkchapman commented on system/jlib/jlzw.cpp:
```diff
@@ -2379,12 +2405,12 @@ ICompressedFileIO *createCompressedFileReader(IFile *file,IExpander *expander, b
                         if (expander&&(trailer.recordSize!=0)) {
```
Feels like this code and that in the method above should be common?

richardkchapman commented:
@jakesmith One comment

jakesmith commented:
@richardkchapman - have commoned up some of the comp. type code. Please check and then I'll squash.

richardkchapman commented:
@jakesmith Looks better

HPCCSmoketest commented:
Automated Smoketest
Sha: b1f7b48328ef37b04db13dfc83c257f29c2e5376
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.475 seconds |

Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unittest result:

| total | passed | failed | timeout | 
|---|---|---|---|
|86 |86 |0 |0 |

HPCC Stop: OK
HPCC Uninstall: OK

jakesmith commented:
@richardkchapman - don't merge, I've spotted a bug.

jakesmith committed:
HPCC-16858 Detect compressed files when reattaching XREF files.

Check physical files, test to see if compressed so as to mark
the meta data correctly, also add size or compressedSize
dependent on whether compressed or not.

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

jakesmith commented:
@richardkchapman - fixed in above commit, also ensure logical file size was set (which was spotted by @cloLN in HPCC-16938).

Please review and I'll squash commits.

mckellyln commented on dali/dfuXRefLib/XRefFilesNode.cpp:
```diff
@@ -360,8 +364,39 @@ bool CXRefFilesNode::AttachPhysical(const char *Partmask,IUserDescriptor* udesc,
+        offset_t physicalSize = iFile->size();
+        bool partCompressed = isCompressedFile(iFile);
+        if (first)
+            isCompressed = partCompressed;
```
Should first be set to false here ?

jakesmith replied:
Yes, it should, will correct.

HPCCSmoketest commented:
Automated Smoketest
Sha: d890936ad9bef49c4b203f7a2791a5f77c91a074
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 90.068 seconds |

Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unittest result:

| total | passed | failed | timeout | 
|---|---|---|---|
|86 |86 |0 |0 |

HPCC Stop: OK
HPCC Uninstall: OK

jakesmith commented:
@mckellyln - added missing first = false, squashed commits and rebsaed on to latest candidate-6.4.0

Please check.

HPCCSmoketest commented:
Automated Smoketest
Sha: d69e608e71f471bbb0228843c64b5c5c7863edb7
Build: failed
Number of Error(s): 5
gmake[4]: *** [tests/libmemcached-1.0/sasl] Error 1
gmake[3]: *** [all] Error 2
make[2]: *** [plugins/memcached/libmemcached-1.0.10/libmemcached/.libs/libmemcached.so.11.0.0] Error 2
make[1]: *** [plugins/memcached/CMakeFiles/generate-libmemcached.dir/all] Error 2
make: *** [all] Error 2

HPCCSmoketest commented:
Automated Smoketest
Sha: d69e608e71f471bbb0228843c64b5c5c7863edb7
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.149 seconds |

Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unittest result:

| total | passed | failed | timeout | 
|---|---|---|---|
|86 |86 |0 |0 |

HPCC Stop: OK
HPCC Uninstall: OK

mckellyln commented:
@jakesmith looks fine.

jakesmith commented:
Thanks.
@richardkchapman - please merge.
